src/app/components/OrderInput.tsx

```diff
@@ -142,73 +175,94 @@ function PositionSizeInput() {
   const validationResult = useAppSelector(validatePositionSize);
   const dispatch = useAppDispatch();
 
+  const [customPercentage, setCustomPercentage] = useState(0);
+  const [isFirstInteraction, setIsFirstInteraction] = useState(true);
```

Comment:
Let's try to not use `useState` since we already have redux, and use only redux for state management.

What is the idea for fist interaction? Why is it something different from any other interaction?

Reply:
> Let's try to not use `useState` since we already have redux, and use only redux for state management.
> 
> What is the idea for fist interaction? Why is it something different from any other interaction?


both these states are the local states of the component. i don't think we should move these state in redux as it is not encouraged [Read Here](https://stackoverflow.com/questions/52156461/should-i-use-redux-when-component-state-is-local) 

Reasons why these two states: 
customPercentage was needed to handle the current percentage selection through which i am changing the color of the active percentage box and also handling the input state of the component.

isFirstInteraction: whenever we load the dexter app it starts showing error on the inputs even though user has not even entered anything yet. therefore, i am keeping is state as a memory for the input field so it will not show error if user has not yet entered anything in that specific input field.

Reply:
About `isFirstInteraction `, I think it's more user-friendly to show the errors if any exist immediately - quickly explains why the submit button is disabled and directs attention to editing them. Maybe instead we can use different defaults, so they would be valid from the start?

OK for `useState` in this case, but (for other cases) I would argue even if the state is local to a component, but requires some network processing or transformation logic - it's better moving them to redux slices, the code will be much more readable easier to test and debug.

Reply:
maybe we need to come up with some other UX for the inputs? because clearly they don't look good the moment you load Dexter and you are welcomed with 2 big red outlined input fields. We should wait for the user to interact with these input fields then we can show them the errors. (That's my thought behind using `isFirstInteraction` ) 

regarding `customPercentage` i was looking for some redux state for current selected percentage but i didn't find any, 
we are only using this `const size = useAppSelector((state) => state.orderInput.size);` state but it returns the calculated size. And since i didn't find any need to move it to redux because we were not using the current selected percentage anywhere else so i made it a local component state. i can move it to redux if you think we might be needing this state somewhere else.

Reply:
> We should wait for the user to interact with these input fields then we can show them the errors. (That's my thought behind using isFirstInteraction )

Yes, I understand why you introduce the `isFirstInteraction`, the alternative suggestion is to leave the validation always active but substitute the defaults with something more reasonable than 0s, so that there wouldn't be an error message right away. 

> maybe we need to come up with some other UX for the inputs?

a related comment in discord - https://discord.com/channels/1125689933735145503/1146445162508197888/1152192758942023721

Reply:
About `customPercentage`, if we have it, we would need to react to `size` input here as well. We can theoretically use only `size` in the state, and when the user types in the size itself, the custom percentage would calculate out of it and get updated also, it would be nice to have, but I don't think it's important to have in the MVP and it will take time to develop and test. I would suggest leaving custom percentage being wiped out (as it is now in the main branch) when user either inputs size or clicks percentage shortcuts, because it's simpler.

Reply:
i will put a placeholder "0"  instead of real value, that way we might be able to get rid of `isFirstInteraction`

Although we are using custom percentage it's just white so we didn't really notice, i made it black in screenshot
![image](https://github.com/DeXter-on-Radix/website/assets/29409407/c3943f2d-bb44-47dc-93c5-129eb0586efc)

we can hide it for now. but this `customPercentage` state is also being used by the percentage buttons. so this state would still be there.

Reply:
It is possible to go without `customPercentage` state if when user licks on shortcuts or enters size (aka amount), the custom percentage input clears, that would simplify the code. 

It's OK to have the `customPercentage`, but if we have it we need to synchronize the size input also - meaning when the user enters the size (amount) itself the custom percentage should also recalculate.

I would even go so far as to remove the shortcuts for the MVP at all - they are buggy and pollute already quite noisy input ui, and we can introduce them again later, more tested and together with the redesign?